### PR TITLE
Auth flow

### PR DESCRIPTION
**Pull request by kennyctran**
@eserpin We should pair this or review since I used conditional rendering of the Dashboard to verify the Auth flow.
@itsme-sunil  @cmcrow2 for the backend PR review.

- Auth flow with Google completed. Can have a conversation with the team if we need more than that.
- Implemented onClick reusability for Add Button
- Abstracted Dashboard views into separate Layouts to conditionally render depending on Auth status
- Created sample Header

Overall, may need to have a quick discussion about Auth since every member, front and back end will need to interact with Auth.

### CODE REVIEW

**Comment by kennyctran** [comment]
Considerably difficult to test if this auth flow works in the Deployment setting since it would take massive customization of secrets to test the preview environment AND the deployment environment. It works on my local environment but at this time unable to confirm if it works in Deployment.

Essentially, the redirect URIs are different for local, preview, and deployment. So I really can only confirm that it works locally, and then change the secrets for deployment which I have but am unable to test until merge with Master.

**Review by cmcrow2 — APPROVED** [review]
Looks great to me!

File: common/layouts/UserDashboardLayout.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Typography from "@material-ui/core/Typography";
+
+export default function UserDashboardLayout({ email }) {
+  return (
+    <>
+      <Typography variant="h2" color="primary">
+        {`Hello, ${email}`}
```
Comment (itsme-sunil) [inline common/layouts/UserDashboardLayout.js:7]: Just wanted to double-check - is this the final decision on how we will refer to a user across the site? By email? Completely fine if it is, just wanted to ensure that that's the front-end decision. 👍

File: package.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "@material-ui/styles": "^4.11.4",
     "axios": "^0.21.1",
     "next": "10.2.2",
+    "next-auth": "^3.23.3",
```
Comment (itsme-sunil) [inline package.json:20]: next-auth looks really cool. Eager to learn more about it

**Review by itsme-sunil — APPROVED** [review]
Looks terrific Kenny, thanks man 👊

**Pull request merged**